packages/core/src/new-api/deploy.ts

```diff
@@ -48,11 +49,14 @@ export async function deploy({
 
   const chainDispatcher = new ChainDispatcherImpl(buildAdaptersFrom(provider));
 
+  const isAutominedNetwork = await checkAutominedNetwork(provider);
```

Comment:
I think we should push this check down into the `deploy` function of `deployer`.

WDYT?

Reply:
After discussion we are happy with the current location. Async in a constructor would be bad.